src/plots/VolumeVTK9/avtVolumeFilter.C

```diff
@@ -0,0 +1,1462 @@
+
+    {
+        debug1 << "CalculateHistogram failed for "
+               << primaryVariable << std::endl;
+    }
+
+    // Get the global histograms acrosss all ranks.
+    std::vector<long long> numvals_out(numValsInHist, 0);
+    SumLongLongArrayAcrossAllProcessors(&(numvals_in[0]),
+                                        &(numvals_out[0]), numValsInHist);
+
+    long long maxVal = 0;
+    for (int i = 0 ; i < numValsInHist ; i++)
+        if (numvals_out[i] > maxVal)
+            maxVal = numvals_out[i];
+
+    std::vector<float> h1(numValsInHist, 0.);
+    if (maxVal != 0)
+    {
+        for (int i = 0 ; i < numValsInHist ; i++)
+            h1[i] = ((double) numvals_out[i]) / ((double) maxVal);
+    }
+
+    MapNode vhist;
+    vhist["histogram_size"] = numValsInHist;
+    vhist["histogram_1d"] = h1;
+
+    visitTimer->StopTimer(t1, "Calculating histogram");
+
+    GetOutput()->GetInfo().GetAttributes().AddPlotInformation("VolumeHistogram", vhist);
+}
+
+// ****************************************************************************
+// Method: avtVolumeFilter::CreateOpacityMap
+//
+// Purpose:
+//   Create the appropriate opacity map.
+//
+// Arguments:
+//
+//
+// Returns:
+//
+// Note:       I factored this out of the RenderImage methods.
+//
+// Programmer: Brad Whitlock
+// Creation:   Tue Aug 22 16:03:09 PDT 2017
+//
+// Modifications:
+//
+// ****************************************************************************
+
+avtOpacityMap
+avtVolumeFilter::CreateOpacityMap(double range[2])
+{
+    unsigned char vtf[4*256];
+    atts.GetTransferFunction(vtf);
+    avtOpacityMap om(256);
+
+    if (atts.GetRendererType() == VolumeAttributes::Composite &&
+             atts.GetSampling()     == VolumeAttributes::Trilinear)
+    {
+        om.SetTableComposite(vtf, 256,
+                             atts.GetOpacityAttenuation() * 2.0 - 1.0,
+                             atts.GetRendererSamples());
+    }
+    else if (atts.GetRendererType() == VolumeAttributes::SLIVR)
+    {
+        // Set the opacity map with opacity correction. This
+        // modifies the opacities though.
+        om.SetTableComposite(vtf, 256,
+                             atts.GetOpacityAttenuation() * 2.0 - 1.0,
+                             atts.GetRendererSamples());
+    }
+    else
+    {
+        // Set the opacity map just using the transfer function.
+        om.SetTable(vtf, 256, atts.GetOpacityAttenuation());
+    }
+
+    double actualRange[2];
+    bool artificialMin = atts.GetUseColorVarMin();
+    bool artificialMax = atts.GetUseColorVarMax();
+    if (!artificialMin || !artificialMax)
+    {
+        GetDataExtents(actualRange, primaryVariable);
+        UnifyMinMax(actualRange, 2);
+    }
+
+    range[0] = (artificialMin ? atts.GetColorVarMin() : actualRange[0]);
+    range[1] = (artificialMax ? atts.GetColorVarMax() : actualRange[1]);
+
+    if (atts.GetScaling() == VolumeAttributes::Log)
+    {
+        if (artificialMin)
```

Comment:
I try to avoid an un-scoped `if` with more than a single line body.

So, either...

```suggestion
        if (artificialMin && range[0] > 0)
            range[0] = log10(range[0]);
```
or

```suggestion
        if (artificialMin)
        {
            if (range[0] > 0)
                range[0] = log10(range[0]);
        }
```

Likewise for companion case below this one.

Reply:
Change made.